### PR TITLE
Feature/api endpoint file restructuring

### PR DESCRIPTION
**Pull request by Rayahhhmed**
1) Fixed the url endpoints. Repurposed database service. Removed database controller endpoints since they are duplicated.
2) Also made a basic 'friend' controller to help Max out when he starts on friend add / remove.
3) Timetable does persist and tested it out too. Just curious if user has only ONE timetable or multiple. I was basically going back and forth and decided to stick to one. 
4) Wanning and I were discussing on the necessity of `timetableId` since we are going to search by userId anyway. This would make no sense if we do not have multiple timetables.

### CODE REVIEW

File: server/package.json [hunk 1]
```diff
@@ -55,6 +55,6 @@
     "@types/node": "17.0.24",
     "prettier": "2.7.1",
     "ts-node": "10.9.1",
-    "typescript": "4.8.2"
+    "typescript": "4.7.4"
```
Comment (munjoonteo) [inline server/package.json:58]: I think you resolved this merge conflict wrong

File: server/src/auth/auth.service.ts [hunk 2]
```diff
@@ -12,15 +16,33 @@ export class AuthService {
     let isCurrentUser = await this.getUser(userInfo.sub);
     if (isCurrentUser === null) {
       const newUser = {
+        // Adding new user information from google to the database
         google_uid: userInfo.sub,
         firstname: userInfo.given_name,
         lastname: userInfo.family_name,
         email: userInfo.email,
         createdAt: new Date().toISOString().slice(0, 10),
+        lastLogin: new Date().toISOString().slice(0, 10),
```
Comment (munjoonteo) [inline server/src/auth/auth.service.ts:25]: Do you think we should keep both the time and date? Could be useful for logging purposes

File: server/src/auth/auth.service.ts [hunk 3]
```diff
@@ -12,15 +16,33 @@ export class AuthService {
...
+        profileURL: userInfo.picture,
+        loggedIn: true,
+        settings: new UserSettingsDto(false, false, false, false, false, false),
```
Comment (munjoonteo) [inline server/src/auth/auth.service.ts:28]: Update the settings DTO with all the new settings we've implemented

File: server/src/auth/auth.service.ts [hunk 4]
```diff
@@ -12,15 +16,33 @@ export class AuthService {
...
       userAdded.save();
       await this.sendEmail(userInfo.email);
+    } else {
+      // Updating the last login date of the user
+      await this.userModel.findOneAndUpdate(
+        { google_uid: userInfo.sub },
+        {
+          $set: {
+            profileURL: userInfo.picture,
+            lastLogin: new Date().toISOString().slice(0, 10),
```
Comment (munjoonteo) [inline server/src/auth/auth.service.ts:41]: Same here

File: server/src/database/database.controller.ts [hunk 5]
```diff
@@ -2,46 +2,20 @@ import {
...
+  Query,
   Request,
   UseInterceptors,
 } from '@nestjs/common';
-import { Timetable, Settings } from 'src/schemas/user.schema';
+import { Timetable, Settings, UserInterface } from 'src/schemas/user.schema';
 import { SessionSerializer } from 'src/auth/session.serializer';
 import { DatabaseService } from './database.service';
 import { UserSettingsDto, UserTimetablesDto } from './dtos/database.dto';
+import { User } from '@sentry/node';
```
Comment (munjoonteo) [inline server/src/database/database.controller.ts:15]: Unwanted import?

File: server/src/database/dtos/database.dto.ts [hunk 6]
```diff
@@ -5,9 +5,34 @@ export class UserSettingsDto {
   isHideFullClasses: boolean;
   isDefaultUnscheduled: boolean;
   isHideClassInfo: boolean;
+
+  constructor(
```
Comment (munjoonteo) [inline server/src/database/dtos/database.dto.ts:9]: This needs to be updated

File: server/src/database/dtos/database.dto.ts [hunk 7]
```diff
@@ -5,9 +5,34 @@ export class UserSettingsDto {
...
+    this.isHideFullClasses = isHideFullClasses;
+    this.isDefaultUnscheduled = isDefaultUnscheduled;
+    this.isHideClassInfo = isHideClassInfo;
+  }
 }
 
 export class UserTimetablesDto {
   selectedCourses: string[];
   selectedClasses: Record<string, Record<string, string>>;
 }
```
Comment (munjoonteo) [inline server/src/database/dtos/database.dto.ts:29]: Add custom events

File: server/src/friend/friend.controller.spec.ts [hunk 8]
```diff
@@ -0,0 +1,18 @@
+import { Test, TestingModule } from '@nestjs/testing';
```
Comment (munjoonteo) [inline server/src/friend/friend.controller.spec.ts:1]: Can we like delete all the test files :  )

File: server/src/friend/friend.controller.ts [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+
+  @Post('/request')
+  async sendFriendRequest(
+    @Request() req,
+    @Body() body: FriendRequestDto,
+  ): Promise<void> {
+    return;
+  }
+
+  @Post('/request')
```
Comment (munjoonteo) [inline server/src/friend/friend.controller.ts:58]: This should be DELETE

File: server/src/friend/friend.controller.ts [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+  UserSettingsDto,
+  UserTimetablesDto,
+} from '../database/dtos/database.dto';
+import { User } from '@sentry/node';
+
+@Controller('friend')
+export class FriendController {
+  constructor(private readonly databaseService: DatabaseService) {}
+
+  @Get('/:userId')
```
Comment (munjoonteo) [inline server/src/friend/friend.controller.ts:26]: All these endpoints should use the LoginGuard

File: server/src/user/user.controller.ts [hunk 11]
```diff
@@ -0,0 +1,93 @@
...
+  UserTimetablesDto,
+} from '../database/dtos/database.dto';
+import { User } from '@sentry/node';
+import { DatabaseService } from 'src/database/database.service';
+
+@Controller('user')
+export class UserController {
+  constructor(private readonly databaseService: DatabaseService) {}
+
+  @Get('/profile/:userId')
```
Comment (munjoonteo) [inline server/src/user/user.controller.ts:26]: All these endpoints should use the login guard

**Review by munjoonteo — CHANGES_REQUESTED** [review]

**Comment by munjoonteo** [comment]
@Rayahhhmed @incalescence When I said 'get rid of the database module' I literally meant delete the database module. There's no way we are yeeting all our actual database-related code in one module. In the friends and user module, since we still need database functionality, you can just import the Mongoose module inside them (as far as I understand) and define a `.service.ts` for each of them. The user setting and timetable DTOs, as well as the friend DTOs, should be in their respective modules.

Also, where's `friend.module.ts`? Also also, you can delete `app.controller.ts` and `app.service.ts` now.

**Comment by Rayahhhmed** [comment]
> 

Yeah sorry, this was my mistake, ill fix it up

**Pull request closed without merging**